core/src/services/upyun/backend.rs

```diff
@@ -0,0 +1,377 @@
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::UpYun)),
+        }?;
+
+        let password = match &self.config.password {
+            Some(password) => Ok(password.clone()),
+            None => Err(Error::new(ErrorKind::ConfigInvalid, "password is empty")
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::UpYun)),
+        }?;
+
+        let client = if let Some(client) = self.http_client.take() {
+            client
+        } else {
+            HttpClient::new().map_err(|err| {
+                err.with_operation("Builder::build")
+                    .with_context("service", Scheme::UpYun)
+            })?
+        };
+
+        let signer = UpYunSigner {
+            operator: operator.clone(),
+            password: password.clone(),
+        };
+
+        Ok(UpYunBackend {
+            core: Arc::new(UpYunCore {
+                root,
+                operator,
+                password,
+                bucket: self.config.bucket.clone(),
+                signer,
+                client,
+            }),
+        })
+    }
+}
+
+/// Backend for upyun services.
+#[derive(Debug, Clone)]
+pub struct UpYunBackend {
+    core: Arc<UpYunCore>,
+}
+
+#[async_trait]
+impl Accessor for UpYunBackend {
+    type Reader = IncomingAsyncBody;
+
+    type BlockingReader = ();
+
+    type Writer = UpYunWriters;
+
+    type BlockingWriter = ();
+
+    type Lister = oio::PageLister<UpYunLister>;
+
+    type BlockingLister = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut am = AccessorInfo::default();
+        am.set_scheme(Scheme::UpYun)
+            .set_root(&self.core.root)
+            .set_native_capability(Capability {
+                stat: true,
+
+                create_dir: true,
+
+                read: true,
+                read_can_next: true,
+
+                write: true,
+                write_can_empty: true,
+                write_can_multi: true,
+                write_with_cache_control: true,
+                write_with_content_type: true,
+
+                write_multi_min_size: Some(1024 * 1024),
+                write_multi_max_size: Some(50 * 1024 * 1024),
+
+                delete: true,
+                rename: true,
+                copy: true,
+
+                list: true,
+                list_with_limit: true,
+                list_with_recursive: true,
+
+                ..Default::default()
+            });
+
+        am
+    }
+
+    async fn create_dir(&self, path: &str, _: OpCreateDir) -> Result<RpCreateDir> {
+        let resp = self.core.create_dir(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => Ok(RpCreateDir::default()),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn rename(&self, from: &str, to: &str, _args: OpRename) -> Result<RpRename> {
+        let resp = self.core.copy(from, to).await?;
```

Comment:
Wrong fn call?

Reply:
copy and rename call the same api in upyun

Reply:
Could you provide a related doc address?

Reply:
https://help.upyun.com/knowledge-base/rest_api/#e7a7bbe58aa8e69687e4bbb6